### PR TITLE
Rename last-assigned-partition-id to last-partition-id in table metadata

### PR DESCRIPTION
**Pull request by jun-he**
Follow up of https://github.com/apache/iceberg/pull/2089

### CODE REVIEW

File: core/src/main/java/org/apache/iceberg/TableMetadataParser.java [hunk 1]
```diff
@@ -291,10 +291,10 @@ static TableMetadata fromJson(FileIO io, InputFile file, JsonNode node) {
           schema, TableMetadata.INITIAL_SPEC_ID, node.get(PARTITION_SPEC)));
     }
 
-    Integer lastAssignedPartitionId = JsonUtil.getIntOrNull(LAST_ASSIGNED_PARTITION_ID, node);
+    Integer lastAssignedPartitionId = JsonUtil.getIntOrNull(LAST_PARTITION_ID, node);
```
Comment (RussellSpitzer) [inline core/src/main/java/org/apache/iceberg/TableMetadataParser.java:294]: Won't this break backwards compatibility?

Comment (rdblue) [reply core/src/main/java/org/apache/iceberg/TableMetadataParser.java:294]: This property was just added, so the rename is safe.

File: core/src/main/java/org/apache/iceberg/BaseUpdatePartitionSpec.java [hunk 2]
```diff
@@ -69,7 +69,7 @@
     this.schema = spec.schema();
     this.nameToField = indexSpecByName(spec);
     this.transformToField = indexSpecByTransform(spec);
-    this.lastAssignedPartitionId = base.lastAssignedPartitionId();
+    this.lastAssignedPartitionId = base.lastPartitionId();
```
Comment (rdblue) [inline core/src/main/java/org/apache/iceberg/BaseUpdatePartitionSpec.java:72]: I don't think we need to rename the method, just the property in the metadata file.

**Comment by rdblue** [comment]
@jun-he, can you roll back the field and and method name changes in other classes? I think we only need to change the table metadata property, not the others.

**Comment by jun-he** [comment]
@rdblue Thanks for the review and updated the PR accordingly.

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks, @jun-he! Looks good.

**Pull request merged**